src/screens/AddSiteScreen.js

```diff
@@ -1,204 +1,339 @@
 import React, {Component} from 'react';
-import { Alert, Picker, ScrollView, Modal, View} from 'react-native';
+import {Alert, Picker, ScrollView, Modal, View} from 'react-native';
+import {connect} from 'react-redux';
 
-import { Button, FormLabel, FormInput, Input, Icon, Overlay, Text } from 'react-native-elements';
+import {Button, FormLabel, FormInput, Input, Icon, Overlay, Text, Card} from 'react-native-elements';
 
-import { campsite, submit_form, common} from '../locale.en';
+import {campsite, submit_form, common} from '../locale.en';
 
-const{ site_description, upload,
-                 campsite_form:{
-                 latitude, longitude,
-                 longitude_placeholder, latitude_placeholder,
-                 add_site_title,site_info,
-                 description,description_placeholder,
-                 directions,directions_placeholder,
-                 nearest_town,nearest_town_placeholder,
-                 here_now, add_site,
-                 accessibility,facilities,
-                 accessibility_options,
-                 facilities_options,}} = campsite;
+const {
+    site_description, upload,
+    campsite_form: {
+        latitude, longitude,
+        longitude_placeholder, latitude_placeholder,
+        add_site_title, site_info,
+        description, description_placeholder,
+        directions, directions_placeholder,
+        nearest_town, nearest_town_placeholder,
+        here_now, add_site,
+        accessibility, facilities,
+        accessibility_options,
+        facilities_options,
+    }
+} = campsite;
 
-const { submit, submitted } = submit_form;
+const {submit, submitted} = submit_form;
 
-const{ title, location } = common;
+const {title, location} = common;
 
-import { navyBlue, grey, darkBlue, lightwhiteblue } from '../styles/index';
-import { CardSection, Card, largeFormElement } from '../components/common/index';
+import {navyBlue, grey, darkBlue} from '../styles/index';
+
+import {
+    updateLatitudeText,
+    updateLongitudeText,
+    updateSiteTitleText,
+    updateSiteDescriptionText,
+    updateSiteDirectionsText,
+    updateSiteNearestTownText,
+    updateAccessibilityOption,
+    updateFacilitiesOption,
+    resetAddScreenFields
+} from '../actions';
 
 class AddSiteScreen extends Component {
-  state = {
-    modalVisible: false,
-  };
+    state = {
+        modalVisible: false,
+    };
 
-  openModal() {
-    this.setState({modalVisible:true});
-  }
+    openModal() {
+        this.setState({modalVisible: true});
+    }
 
-  closeModal() {
-    this.setState({modalVisible:false});
-  }
+    closeModal() {
+        this.setState({modalVisible: false});
+    }
```

Comment:
Damn, I meant to get this and totally forgot. @amgoodman5 we should refactor away from using the component-level state and have this logic be passed around via redux. With the actions and reducers already in place, it should be straightforward.

Reply:
👍

Reply:
Good point.

Reply:
When using you build, do you use local state first then hook it up to redux or start in redux? Let me know if that question makes sense.

Reply:
Yea, just try to go with redux first, and as you get the hang of it it'll come together naturally. I always go that route lately instead of falling back to local component state.